visualization/app/codeCharta/ui/loadingGif/loadingGif.component.ts

```diff
@@ -0,0 +1,65 @@
+import "./loadingGif.component.scss"
+import {IRootScopeService, ITimeoutService} from "angular";
+
+export interface LoadingGifComponentSubscriber {
+    onLoadingFileStatusChanged(isLoadingFile: boolean, event: angular.IAngularEvent)
+    onLoadingMapStatusChanged(isLoadingMap: boolean, event: angular.IAngularEvent)
+}
+
+export class LoadingGifController implements LoadingGifComponentSubscriber {
+
+    public static readonly LOADING_FILE_STATUS_EVENT = "loading-file-status-changed"
+    public static readonly LOADING_MAP_STATUS_EVENT = "loading-map-status-changed"
+
+    private _viewModel: {
+        isLoadingFile: boolean,
+        isLoadingMap: boolean
+    } = {
+        isLoadingFile: true,
+        isLoadingMap: true
+    }
+
+    /* @ngInject */
+    constructor(
+        private $rootScope: IRootScopeService,
+        private $timeout: ITimeoutService
+    ){
+        LoadingGifController.subscribe(this.$rootScope, this)
```

Comment:
The LoadingGifSubscriber is only used here. In that case, we could also make methods such as "onLoadingFileStatusChanged" static and don't use angular events at all

Reply:
I actually prefer sending events from the other components than calling the loadingGif component directly, but if you have a good idea, tell me what you think of ;)

Reply:
now handled with a loadingGif.service